kolibri/plugins/setup_wizard/assets/src/machines/wizardMachine.js

```diff
@@ -239,23 +239,10 @@ export const wizardMachine = createMachine(
           BACK: 'requirePassword',
         },
       },
-      // A passthrough step depending on the value of context.canGetOsUser - the finalizeSetup state
-      // will provision the device with the OS user and create the default facility
-      createSuperuserAndFacility: {
-        on: { BACK: 'personalDataConsent' },
-        always: [
-          {
-            cond: 'canGetOsUser',
-            target: 'finalizeSetup',
-          },
-          {
-            target: 'createSuperuserAndFacilityForm',
-          },
-        ],
-      },
 
-      // If we're not able to get an OS user, the user creates their account
-      createSuperuserAndFacilityForm: {
+      // In the group learning flow we always create the account on the device
+      // and the backend associates it with the created superuser.
```

Comment:
I'm wondering if maybe we want to keep the passthrough but , for the first object in the `always` array, we check both for `canGetOsUser` and `isOnMyOwnOrGroup` (which should probably be renamed `isOnMyOwn` because it's a boolean). I think you can `cond: ['canGetOsUser', 'isOnMyOwnOrGroup'],` to apply multiple conditions there.

This assumes that the "On My Own" flow when we have the OS User should still go from personal data consent to "here is your Kolibri" though. 

This should allow us to show the form on the group flow while still giving that expedited provisioning experience.

Reply:
I am not sure I understand the motivation here - this state is not used at all for the 'on my own' setup flow, it is only used for the 'group learning' flow - the conditional behaviour for the 'on my own' setup is handled in this state: https://github.com/learningequality/kolibri/blob/2644c03a714ceb1448bb537ef3801a3eb6b960ab/kolibri/plugins/setup_wizard/assets/src/machines/wizardMachine.js#L124

Reply:
Ahh you're right! I thought I followed the right path in the diff 😵‍💫